### PR TITLE
Use dynamic throw type extension on EntityManagerInterface::flush

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

File: src/Type/Doctrine/EntityManagerInterfaceThrowTypeExtension.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+	public function getThrowTypeFromMethodCall(MethodReflection $methodReflection, MethodCall $methodCall, Scope $scope): ?Type
+	{
+		$type = $scope->getType($methodCall->var);
+
+		if ((new ObjectType(EntityManagerInterface::class))->isSuperTypeOf($type)->yes()) {
+			return TypeCombinator::union(
+				...array_map(static function ($class): Type {
+					return new ObjectType($class);
+				}, self::SUPPORTED_METHOD[$methodReflection->getName()])
```
Comment (ondrejmirtes) [inline src/Type/Doctrine/EntityManagerInterfaceThrowTypeExtension.php:42]: This needs a test. You can create a new test case for the corresponding rule, and fetch the rule with `self::getContainer()->getByType(....`

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Type/Doctrine/EntityManagerInterfaceThrowTypeExtension.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Type/Doctrine/EntityManagerInterfaceThrowTypeExtension.php:42]: This seems covered by 
https://github.com/phpstan/phpstan-doctrine/blob/1.3.x/tests/Rules/Exceptions/data/unthrown-exception.php#L21-L25
to me.

I tried to comment the whole if in the `getThrowTypeFromMethodCall`, and tests were failing.

Comment (ondrejmirtes) [reply src/Type/Doctrine/EntityManagerInterfaceThrowTypeExtension.php:42]: If your `src/` does not change anything about test results then it's not needed.

Please show why your change is needed by writing a failing test first.

Comment (VincentLanglet) [reply src/Type/Doctrine/EntityManagerInterfaceThrowTypeExtension.php:42]: I added another test, which fail without the change

Comment (VincentLanglet) [reply src/Type/Doctrine/EntityManagerInterfaceThrowTypeExtension.php:42]: I added another test which fail without this change.

**Review by ondrejmirtes — APPROVED** [review]

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**